src/wallet/wallet.cpp

```diff
@@ -1464,6 +1476,22 @@ void CWallet::transactionRemovedFromMempool(const CTransactionRef& tx, const Mem
     if (it != mapWallet.end()) {
         RefreshMempoolStatus(it->second, chain());
     }
+
+    auto* replaced_reason = std::get_if<ReplacedReason>(&reason);
+
+    // Check if wallet transaction is being replaced by a parent transaction which is not from this wallet
+    if (IsFromMe(*tx) && replaced_reason != nullptr && replaced_reason->replacement_tx.has_value() && !IsFromMe(*replaced_reason->replacement_tx.value())) {
+        m_unrelated_conflict_tx_watchlist.insert(std::make_pair(replaced_reason->replacement_tx.value()->GetHash(), tx));
+    }
+
+    auto iter = m_unrelated_conflict_tx_watchlist.find(tx->GetHash());
+    if (iter != m_unrelated_conflict_tx_watchlist.end()) {
+        // The replacement tx was removed from the mempool, remove it from map
+        // This new replacement tx may not conflict with the original tx
+        // so leave wallet tx to remain as TxStateInactive
+        m_unrelated_conflict_tx_watchlist.erase(iter);
+    }
```

Comment:
In 71200b72ab848c3066e5123dc0badc24c9648f47 "Handle double-spending of unrelated parents to wallet txs"
```suggestion
    auto* replaced_reason = std::get_if<ReplacedReason>(&reason);
    if (IsFromMe(*tx) && replaced_reason != nullptr) {

        // Check if wallet transaction is being replaced by a parent transaction which is not from this wallet
        if (replaced_reason->replacement_tx.has_value() && !IsFromMe(*replaced_reason->replacement_tx.value())) {
            m_unrelated_conflict_tx_watchlist.insert(std::make_pair(replaced_reason->replacement_tx.value()->GetHash(), tx));
        }
        // Remove the replacement tx was removed from the mempool, remove it from map
        // This new replacement tx may not conflict with the original tx
        // so leave wallet tx to remain as TxStateInactive
        m_unrelated_conflict_tx_watchlist.erase(tx->GetHash());
    }
```